File: backend/src/models/Image.ts
import { Entity, Column, PrimaryGeneratedColumn, ManyToOne, JoinColumn } from 'typeorm';
import Orphanage from './Orphanage';

@Entity({ name: 'images' })
export default class Image {
  @PrimaryGeneratedColumn('increment')
  id: number;

  @Column()
  name: string;

  @ManyToOne(type => Orphanage, orphanage => orphanage.images)
  @JoinColumn({ name: 'orphanage_id' })
  orphanage: Orphanage;

  constructor(name: string) {
    this.name = name;
  }
}

File: mobile/lib/models/Image.dart
class Image {
  int id;
  String name;

  Image(int id, String name) {
    this.id = id;
    this.name = name;
  }

  @override
  String toString() {
    return '''
      ${super.toString()},
      id: $id,
      name: $name
    ''';
  }
}
